### DISCUSSION TITLE
Only returning 1 row of data using PostgreSQL

### DISCUSSION THREAD

**Discussion by starmomou** [Budibase/budibase#3447] [Help] [answered]
Hi, I'm using the self-hosted Budibase with PostgreSQL as my data source. I remembered once I saw someone has raised a similar issue in discussion, but I could not find it back. Is that a bug? 

**Version**
PostgreSQL`14.1`
Budibase `0.9.187`

<img width="1329" alt="螢幕截圖 2021-11-19 下午4 02 21" src="https://user-images.githubusercontent.com/73826858/142588942-b6664ab7-f8dc-47a2-8cc9-55f7049e02cf.png">

**Comment by mjashanks** [comment, marked as answer]
Hi @starmomou. I think this is because your table does not have a primary key.

**Reply by starmomou** [reply]
Thanks!
